robotiq_c_model_control/src/robotiq_c_model_control/RobotiqGripperActionInterface.py

```diff
@@ -0,0 +1,331 @@
+                if status.gSTA == 0:
+                    break
+
+            if rospy.is_shutdown():
+                self.resetting = False
+                self.has_goal = False
+                return
+
+        # Active Gripper
+        goal = CModel_robot_output()
+        goal.rACT = 1
+        self.send_gripper_command(goal, parse=False)
+        while not rospy.is_shutdown() and not self.interrupted:
+            with self.status_lock:
+                status = deepcopy(self.last_status)
+            if status.gSTA == 3:
+                break
+
+        if rospy.is_shutdown():
+            self.resetting = False
+            self.has_goal = False
+            return
+
+        rospy.sleep(1.0)
+        # Get Actual Calibrated Values
+        goal = GripperGoal()
+        goal.force = 255
+        goal.direction = goal.CLOSE
+        goal.auto_release = goal.DISABLED
+        if not self.send_gripper_command(goal):
+            self.has_goal = False
+            self.resetting = False
+            rospy.signal_shutdown('Failed Close Calibration')
+            return
+
+        rospy.sleep(1.0)
+        with self.status_lock:
+            self.max_closed = self.last_status.gPO
+
+        goal = GripperGoal()
+        goal.force = 255
+        goal.direction = goal.OPEN
+        goal.auto_release = goal.DISABLED
+        if not self.send_gripper_command(goal):
+            self.has_goal = False
+            self.resetting = False
+            rospy.signal_shutdown('Failed Open Calibration')
+            return
+
+        rospy.sleep(1.0)
+        with self.status_lock:
+            self.max_open = self.last_status.gPO
+
+        self.has_goal = False
+        self.resetting = False
+
+    def parse_cmd(self, goal):
+        cmd = CModel_robot_output()
+
+        # Auto Release
+        if goal.auto_release == goal.ENABLED:
+            cmd.rATR = 1
+            cmd.rADR = 1 if goal.auto_release_direction == goal.OPEN else 0
+        else:
+            cmd.rATR = 0
+
+        # Set active and go to
+        cmd.rACT = 1
+        cmd.rGTO = 1
+
+        # Position
+        cmd.rPR = self.map_position_to_calibration(goal)
+
+        # Speed - Always set to max for now
+        cmd.rSP = 255
+
+        # Force
+        if goal.force > 255:
+            cmd.rFR = 255
+        elif goal.force < 0:
+            cmd.rFR = 0
+        else:
+            cmd.rFR = int(goal.force)
+
+        self.command_pub.publish(cmd)
+        return cmd
+
+    def send_gripper_command(self, goal, parse=True):
+        try:
+            if parse:
+                goal = self.parse_cmd(goal)
+            self.send_command(goal)
+        except Exception as exc:
+            self.__log.err('Error While Sending Command [{}] - Shutting Down'.format(exc))
+            rospy.signal_shutdown('Error While Sending Command - Shutting Down')
+            return False
+        return True
+
+    @staticmethod
+    def get_fault_text(fault_code):
```

Comment:
There's a whole lotta branchin' going on here.

Wouldn't this be easier to read if it were a dict?

Reply:
fixed